test/e2e/plain_provisioner_test.go

```diff
@@ -544,6 +544,130 @@ var _ = Describe("plain provisioner bundleinstance", func() {
+				if err := c.Get(ctx, client.ObjectKeyFromObject(bi), bi); err != nil {
+					return nil, err
+				}
+				return meta.FindStatusCondition(bi.Status.Conditions, rukpakv1alpha1.TypeInstalled), nil
+			}).Should(And(
+				Not(BeNil()),
+				WithTransform(func(c *metav1.Condition) string { return c.Type }, Equal(rukpakv1alpha1.TypeInstalled)),
+				WithTransform(func(c *metav1.Condition) metav1.ConditionStatus { return c.Status }, Equal(metav1.ConditionFalse)),
+				WithTransform(func(c *metav1.Condition) string { return c.Reason }, Equal(rukpakv1alpha1.ReasonInstallFailed)),
+				WithTransform(func(c *metav1.Condition) string { return c.Message }, ContainSubstring(`uninstall failed`)),
+			))
+		})
+		When("the dependent BundleInstance has been created", func() {
+			var (
+				providesBundle *rukpakv1alpha1.Bundle
+				providesBI     *rukpakv1alpha1.BundleInstance
+			)
+			BeforeEach(func() {
+				ctx = context.Background()
+
+				By("creating the provides Bundle resource")
+				providesBundle = &rukpakv1alpha1.Bundle{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "e2e-bundle-provides-",
+					},
+					Spec: rukpakv1alpha1.BundleSpec{
+						ProvisionerClassName: plainProvisionerID,
+						Source: rukpakv1alpha1.BundleSource{
+							Type: rukpakv1alpha1.SourceTypeImage,
+							Image: &rukpakv1alpha1.ImageSource{
+								Ref: "testdata/bundles/plain-v0:provides",
+							},
+						},
+					},
+				}
+				err := c.Create(ctx, providesBundle)
+				Expect(err).To(BeNil())
+
+				By("creating the provides BI resource")
+				providesBI = &rukpakv1alpha1.BundleInstance{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "e2e-bi-provides-",
+					},
+					Spec: rukpakv1alpha1.BundleInstanceSpec{
+						ProvisionerClassName: plainProvisionerID,
+						BundleName:           providesBundle.GetName(),
+					},
+				}
+				err = c.Create(ctx, providesBI)
+				Expect(err).To(BeNil())
+			})
+			AfterEach(func() {
+				By("deleting the testing provides Bundle resource")
+				Expect(client.IgnoreNotFound(c.Delete(ctx, providesBundle))).To(BeNil())
+
+				By("deleting the testing provides BundleInstance resource")
+				Expect(client.IgnoreNotFound(c.Delete(ctx, providesBI))).To(BeNil())
+
+			})
+			It("should eventually project a successful installation", func() {
+				Eventually(func() (*metav1.Condition, error) {
+					if err := c.Get(ctx, client.ObjectKeyFromObject(bi), bi); err != nil {
+						return nil, err
+					}
+					return meta.FindStatusCondition(bi.Status.Conditions, rukpakv1alpha1.TypeInstalled), nil
+				}).Should(And(
+					Not(BeNil()),
+					WithTransform(func(c *metav1.Condition) string { return c.Type }, Equal(rukpakv1alpha1.TypeInstalled)),
+					WithTransform(func(c *metav1.Condition) metav1.ConditionStatus { return c.Status }, Equal(metav1.ConditionTrue)),
+					WithTransform(func(c *metav1.Condition) string { return c.Reason }, Equal(rukpakv1alpha1.ReasonInstallationSucceeded)),
```

Comment:
Should we explicitly check for an empty message? Not sure if we do this in other places, but maybe we should. That would add a guarantee that we don't accidentally leave a stale message in the status once an install succeeds